cpp/oneapi/dal/algo/rbf_kernel/backend/gpu/compute_kernel_dense_dpc.cpp

```diff
@@ -29,46 +27,112 @@ using input_t = compute_input<task::compute>;
 using result_t = compute_result<task::compute>;
 using descriptor_t = detail::descriptor_base<task::compute>;
 
-namespace daal_rbf_kernel = daal::algorithms::kernel_function::rbf;
-namespace interop = dal::backend::interop;
+namespace pr = dal::backend::primitives;
+
+template <typename Float>
+inline auto compute_rbf_values(sycl::queue& queue,
+                               const pr::ndview<Float, 1>& sqr_x_nd,
+                               const pr::ndview<Float, 1>& sqr_y_nd,
+                               pr::ndview<Float, 2>& res_nd,
+                               double sigma,
+                               const dal::backend::event_vector& deps = {}) {
+    const std::int64_t x_row_count = sqr_x_nd.get_dimension(0);
+    const std::int64_t y_row_count = sqr_y_nd.get_dimension(0);
+    ONEDAL_ASSERT(res_nd.get_count() == x_row_count * y_row_count);
+
+    const Float coeff = static_cast<Float>(-0.5 / (sigma * sigma));
+
+    const Float* sqr_x_ptr = sqr_x_nd.get_data();
+    const Float* sqr_y_ptr = sqr_y_nd.get_data();
+    Float* res_ptr = res_nd.get_mutable_data();
+
+    const Float threshold = dal::backend::exp_low_threshold<Float>();
+
+    const auto wg_size = dal::backend::propose_wg_size(queue);
+    const auto range =
+        dal::backend::make_multiple_nd_range_2d({ x_row_count, y_row_count }, { wg_size, 1 });
+
+    const std::size_t ld = y_row_count;
+
+    auto compute_rbf_event = queue.submit([&](sycl::handler& cgh) {
+        cgh.depends_on(deps);
+
+        cgh.parallel_for(range, [=](sycl::nd_item<2> item) {
+            const std::size_t i = item.get_global_id(0);
+            const std::size_t j = item.get_global_id(1);
+            const Float sqr_x_i = sqr_x_ptr[i];
+            const Float sqr_y_j = sqr_y_ptr[j];
+            const Float res_rbf_ij = res_ptr[i * ld + j];
+            const Float arg = sycl::fmax((sqr_x_i + sqr_y_j + res_rbf_ij) * coeff, threshold);
+
+            res_ptr[i * ld + j] = sycl::exp(arg);
+        });
+    });
+
+    return compute_rbf_event;
+}
 
 template <typename Float>
-using daal_rbf_kernel_t =
-    daal_rbf_kernel::internal::KernelImplRBFOneAPI<daal_rbf_kernel::defaultDense, Float>;
+inline auto compute_rbf(sycl::queue& queue,
+                        const pr::ndview<Float, 2>& x_nd,
+                        const pr::ndview<Float, 2>& y_nd,
+                        pr::ndview<Float, 2>& res_nd,
+                        double sigma,
+                        const dal::backend::event_vector& deps = {}) {
+    const std::int64_t x_row_count = x_nd.get_dimension(0);
+    const std::int64_t y_row_count = y_nd.get_dimension(0);
+
+    auto sqr_x_nd = pr::ndarray<Float, 1>::empty(queue, { x_row_count }, sycl::usm::alloc::device);
+    auto sqr_y_nd = pr::ndarray<Float, 1>::empty(queue, { y_row_count }, sycl::usm::alloc::device);
+
+    auto reduce_x_event =
+        reduce_by_rows(queue, x_nd, sqr_x_nd, pr::sum<Float>{}, pr::square<Float>{}, deps);
```

Comment:
How does it work? `reduce_by_rows` is in `pr` namespace, do we have `using namespace`?